### ISSUE TITLE
Release v1.0.0

### ISSUE THREAD

**Issue by wavebeem** [jneen/parsimmon#105]
Parsimmon has been API stable for long enough---I don't think there are any necessary breaking changes---that we should just release v1.0.0 already. I want to double check there isn't anything we ought to remove before doing this important bump, though.

**Comment by leonardfactory** [comment]
I'd be glad to help during this important phase. If there's some area where you'd like a contributor to explore for eventual changes/removals, let me know.

PS: I've been using Parsimmon for a while right now, and it's a beautiful library. Thanks!

**Comment by wavebeem** [comment]
Hi @leonardfactory, thanks for the kind words.

The most important thing for me right now is just double-checking that nothing in the library is totally unnecessary, I suppose. And it might be a good time to add a new function or two, if anything good comes to mind.

The thing I'm most interested in dropping is the currently exposed `Parsimmon.Parser` class. I can see that someone might want to check `if (x instanceof Parsimmon.Parser)`, but it's just weird to expose that in general because it makes people think they should use `new Parsimmon.Parser` even though that's not right. I think replacing that export with an `isParser` function would make a lot of sense.

Also I've been thinking of a function that's a shortcut for `.or(Parsimmon.of(foo))`, but I can't decide what to call it.

It might be nice to have a GitHub page for the library too, but it's definitely more work to maintain that vs a README file, so it's only a maybe.

Anyway, happy to hear your thoughts on the situation.

**Comment by leonardfactory** [comment]
@wavebeem 
> Also I've been thinking of a function that's a shortcut for .or(Parsimmon.of(foo)), but I can't decide what to call it.
I was thinking about `orJust`, i.e.:

```js
var parser = Parsimmon.string('optional').orJust('nothing');
```

Imho it resembles the functional `Just` and seems appropriate. What do you think?
I'll start working on it soon 😄

**Comment by wavebeem** [comment]
My only complaint about `orJust` is that we aren't using it anywhere else, nor any Maybe monad. I think `.else` sounds really good!

Yeah but since ES5 you can use keywords as method calls anyway, like `foo.else(bar)` instead of the clunky `foo['else'](bar)`, so I don't think it's a big deal. We could always include an alias like `else_` for ES3. Could also be `.orElse`, but it's a bit wordy.

**Comment by wavebeem** [comment]
I'm gonna chew on the name for a little bit, but feel free to work on the PR if you have time and I can just comment on it with my thoughts on the name.

**Comment by leonardfactory** [comment]
I think it's done and I'd appreciate your review! Talking about the name, `default` seems another nice candidate. Let me know your thoughts.

**Comment by wavebeem** [comment]
Oops, I already did this and forgot to close the issue…
